inc/saiport.h

```diff
@@ -431,6 +433,14 @@ typedef enum _sai_port_attr_t
      * given port list will be dropped. */
     SAI_PORT_ATTR_EGRESS_BLOCK_PORT_LIST,
 
+    /** Dropping of 802.1BR untagged frames on ingress [bool] (default to FALSE).
+     * Applicable only to Physical ports. */
+    SAI_PORT_ATTR_DOT1BR_DROP_UNTAGGED,
```

Comment:
Dell - Internal Use - Confidential
Hi Atit,

Yes. It is possible to derive the ‘drop etag tagged/untagged’ parameter from the br type (cascade, upstream, access). But there could be some violation scenarios, where receiving a particular type of frame on a port is invalid. For example, receiving dot1br Tagged frames might be invalid on dot1br Access port. Similarly receiving dot1br Untagged frames might be invalid on cascading port. It all depends on the SAI application’s need. We have some proprietary SAI applications in our software with these requirements.

Also on a non-dot1br port (port that is neither dot1br Access/Cascading/Upstream type), dot1br tagged frames should be dropped. Receiving dot1br frames on a non-dot1br port indicates a mis-configuration in the network.

Regards,
Ravi

From: Atit Jain [mailto:notifications@github.com]
Sent: Monday, June 20, 2016 6:10 PM
To: opencomputeproject/SAI <SAI@noreply.github.com>
Cc: Sivasankar, Ravikumar <Ravikumar_Sivasankar@Dell.com>; Author <author@noreply.github.com>
Subject: Re: [opencomputeproject/SAI] IEEE 802.1BR SAI proposal. (#73)


In inc/saiport.h<https://github.com/opencomputeproject/SAI/pull/73#discussion_r67682226>:

> @@ -431,6 +433,14 @@ typedef enum _sai_port_attr_t

>       * given port list will be dropped. */

>      SAI_PORT_ATTR_EGRESS_BLOCK_PORT_LIST,

>

> +    /** Dropping of 802.1BR untagged frames on ingress [bool] (default to FALSE).

> +     * Applicable only to Physical ports. */

> +    SAI_PORT_ATTR_DOT1BR_DROP_UNTAGGED,

How these attributes interact with dot1br and dot1br_extended port objects. If a given port is set as physical port for one of the br types (cascade, upstream, access) , we can use the br type as a hint to derive drop etag tagged/untagged. Is there some other use case I am missing?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/opencomputeproject/SAI/pull/73/files/279f0335c33c0c831ec955e1cf93daf61adf52d5#r67682226>, or mute the thread<https://github.com/notifications/unsubscribe/AOoHu4wVYWW-TU-5xs2zR5z008DliWyPks5qNooOgaJpZM4HpSGE>.